### ISSUE TITLE
Proguard removes com.google.android.datatransport.Priority

### ISSUE THREAD

**Issue by tpucci** [firebase/firebase-android-sdk#2441]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 4.1.2
  * Firebase Component:
 ```groovy
	  firebase_analytics                : "18.0.0",
	  firebase_auth                     : "20.0.2",
	  firebase_crashlytics              : "17.3.0",
	  firebase_crashlytics_gradle_plugin: "2.5.0",
	  firebase_dynamic_links            : "19.1.1",
	  firebase_messaging                : "21.0.1",
	  firebase_remote_config            : "20.0.1",
```

### [REQUIRED] Step 3: Describe the problem

The app freezes but do not crash.
Indeed, some class is not found at the time the crash report is sent.

It is the same issue as #1996.
In the issue, it is said that the fix has been released.
However, my project resolves `com.google.android.datatransport:transport-api:2.2.1` and does not contain the fix commit https://github.com/firebase/firebase-android-sdk/commit/690d119e6534708180b27942529db4cfd544b7c9

#### Steps to reproduce:

- build in release mode with Proguard enabled
- crash the app

#### Relevant Code:

```
java.lang.NoClassDefFoundError: c.e.a.a.i.x.a
    at c.e.a.a.i.x.a.a(PriorityMapping.java:1)
    at c.e.a.a.i.v.h.A.i(SQLiteEventStore.java:4)
    at c.e.a.a.i.v.h.A.p(SQLiteEventStore.java:5)
    at c.e.a.a.i.v.h.v.apply(Unknown Source:8)
    at c.e.a.a.i.v.h.A.k(SQLiteEventStore.java:3)
    at c.e.a.a.i.v.h.A.K0(SQLiteEventStore.java:6)
    at c.e.a.a.i.v.c.b(DefaultScheduler.java:1)
    at c.e.a.a.i.v.b.execute(Unknown Source:6)
    at c.e.a.a.i.v.h.A.a(SQLiteEventStore.java:3)
    at c.e.a.a.i.v.c.c(DefaultScheduler.java:8)
    at c.e.a.a.i.v.a.run(Unknown Source:8)
    at c.e.a.a.i.i$a.run(SafeLoggingExecutor.java:1)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.ExceptionInInitializerError
    ... 15 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: c.e.a.a.d.values []
    at java.lang.Enum.enumValues(Enum.java:270)
    at java.lang.Enum.access$000(Enum.java:61)
    at java.lang.Enum$1.create(Enum.java:277)
    at java.lang.Enum$1.create(Enum.java:275)
    at libcore.util.BasicLruCache.get(BasicLruCache.java:63)
    at java.lang.Enum.getSharedConstants(Enum.java:289)
    at java.lang.Class.getEnumConstantsShared(Class.java:2428)
    at java.util.EnumMap.getKeyUniverse(EnumMap.java:755)
    at java.util.EnumMap.<init>(EnumMap.java:138)
    at c.e.a.a.i.x.a.<clinit>(PriorityMapping.java:2)
    ... 15 more
Caused by: java.lang.NoSuchMethodException: c.e.a.a.d.values []
    at java.lang.Class.getMethod(Class.java:2072)
    at java.lang.Class.getDeclaredMethod(Class.java:2050)
    at java.lang.Enum.enumValues(Enum.java:267)
    ... 24 more
```

### Workaround

See #1996:
```
-keepclassmembers enum com.google.android.datatransport.Priority {
    public static **[] values();
}
```

**Comment by Odaym** [comment]
happens here too, workaround does work

**Comment by mrichards** [comment]
It looks like you have the latest version of the `com.google.android.datatransport:transport-api` artifact (2.2.1). Check your dependencies for `com.google.firebase:firebase-datatransport`? The fix for your issue is in 17.0.9+, and the latest is 17.0.11.

I'm surprised Gradle didn't resolve the latest version for you, though updating all your Firebase SDKs to their latest versions should resolve the issue. (Using the [BoM](https://firebase.google.com/docs/android/learn-more#bom) makes it easy to get all the latest Firebase SDK versions with one change!) Or you can force Gradle to take a specific firebase-datatransport version by declaring it explicitly with `force` or `strictly`.

Please reopen the issue if the problem persists after switching to the latest `firebase-datatransport` dependency.

**Comment by tpucci** [comment]
Will try, thanks!
